### PR TITLE
doc: README.rst added to subsys/console/echo sample

### PR DESCRIPTION
**Pull request by Spice-Weasel**
README added to the console echo sample

### CODE REVIEW

File: samples/subsys/console/echo/README.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.. zephyr:code-sample:: console_echo
+   :name: console echo
+   :relevant-api: console_api
+
+   Use console_getchar() and console_putchar() to echo an input character back
+   to the output.
```
Comment (kartben) [inline samples/subsys/console/echo/README.rst:6]: Focus on end result:

```suggestion
   Echo an input character back to the output using the Console API.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/subsys/console/echo/README.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Overview
+********
+
+This example shows how the :c:func:`console_getchar` function and
+:c:func:`console_putchar` function can be used to echo an input character
+to a UART console.
```
[the remark this answers is not in the document]
Comment (Spice-Weasel) [reply samples/subsys/console/echo/README.rst:13]: Good point, I'll make it more generic - so I suppose `usart2` could be something else entirely?

`	model = "STMicroelectronics STM32F401RE-NUCLEO board";
	compatible = "st,stm32f401re-nucleo";

	chosen {
		zephyr,console = &usart2;
		zephyr,shell-uart = &usart2;
		zephyr,sram = &sram0;
		zephyr,flash = &flash0;
		zephyr,code-partition = &slot0_partition;
	};`

File: include/zephyr/console/console.h [hunk 3]
```diff
@@ -15,6 +15,13 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Console API
+ * @defgroup console_api Console API
+ * @ingroup os_services
+ * @{
```
Comment (kartben) [inline include/zephyr/console/console.h:22]: missing closing curly brace?

Comment (Spice-Weasel) [reply include/zephyr/console/console.h:22]: Thanks, I haven't worked with doxygen before.

File: samples/subsys/console/echo/README.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
+.. zephyr:code-sample:: console_echo
+   :name: console echo
+   :relevant-api: console_api
```
[the remark this answers is not in the document]
Comment (kartben) [reply samples/subsys/console/echo/README.rst:3]: You would still need to add a new page under https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/index.rst
It's OK if for now it only contains the API Reference entry, similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/poweroff.rst

[the remark this answers is not in the document]
Comment (Spice-Weasel) [reply samples/subsys/console/echo/README.rst:3]: Oh right, ok - I'll add a page. Thanks Ben!

[the remark this answers is not in the document]
Comment (Spice-Weasel) [reply samples/subsys/console/echo/README.rst:3]: Should be ok now. I'm not sure to inspect the build to ensure it has rendered correctly however.

[the remark this answers is not in the document]
Comment (kartben) [reply samples/subsys/console/echo/README.rst:3]: > Should be ok now. I'm not sure to inspect the build to ensure it has rendered correctly however.

Looking good, yes :) 

https://builds.zephyrproject.io/zephyr/pr/64626/docs/services/console.html
https://builds.zephyrproject.io/zephyr/pr/64626/docs/samples/subsys/console/echo/README.html#console_echo

File: samples/subsys/console/echo/README.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
+.. zephyr:code-sample:: console_echo
+   :name: console echo
```
Comment (kartben) [inline samples/subsys/console/echo/README.rst:2]:
```suggestion
   :name: Console echo
```

File: samples/subsys/console/echo/README.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+:zephyr:code-sample:`console_getchar`.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/console/echo
+   :host-os: unix
+   :board: nucleo_f401re
+   :goals: build flash
+   :compact:
+
+Following the initial prompt given by the firmware start pressing keys on a
```
Comment (kartben) [inline samples/subsys/console/echo/README.rst:33]:
```suggestion
Following the initial prompt given by the firmware, start pressing keys on a
```

File: samples/subsys/console/echo/README.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+   :name: console echo
+   :relevant-api: console_api
+
+   Echo an input character back to the output using the Console API.
+
+Overview
+********
+
+This example shows how the :c:func:`console_getchar` function and
+:c:func:`console_putchar` function can be used to echo an input character
```
Comment (kartben) [inline samples/subsys/console/echo/README.rst:11]:
```suggestion
:c:func:`console_putchar` functions can be used to echo an input character
```

File: samples/subsys/console/echo/README.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+back to the console.
+
+Requirements
+************
+
+UART console is required to run this sample.
```
Comment (kartben) [inline samples/subsys/console/echo/README.rst:17]: Similar to an earlier comment regarding being unnecessarily uart-specific, isn't the requirement rather: "A board that defines a `zephyr,console`."?

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/subsys/console/echo/README.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+********
+
+This example shows how the :c:func:`console_getchar` and
+:c:func:`console_putchar` functions can be used to echo an input character
+back to the console.
+
+Requirements
+************
+
+UART console is required to run this sample.
```
Comment (Spice-Weasel) [inline samples/subsys/console/echo/README.rst:17]: @kartben I saw you mentioned this should change - I am perhaps referring mistakenly to the terminal program I am using here (picocom). This is also consistent with the other console samples.

Comment (kartben) [reply samples/subsys/console/echo/README.rst:17]: Ya I removed my comment as I was mostly getting confused between shell and console :) I don't think a console that's *not* UART makes sense so nevermind :)

**Review by kartben — APPROVED** [review]
Thank you very much!

**Comment by Spice-Weasel** [comment]
Morning, can someone else review this please?

**Comment by Spice-Weasel** [comment]
> Morning, can someone else review this please?

@nashif @carlescufi @kartben

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**